pkg/certmanager/certmanager.go

```diff
@@ -121,7 +121,7 @@ func (c *CertManager) EnsureCertificate(vol *csiapi.MetaData, keyBundle *util.Ke
 	glog.Infof("cert-manager: certificate written to file %s", certPath)
 
 	keyPath := util.KeyPath(vol)
-	if err := util.WriteFile(keyPath, keyBundle.PEM, 0600); err != nil {
+	if err := util.WriteFile(keyPath, keyBundle.PEM, 0644); err != nil {
```

Comment:
Some software will trip up with these permissions I think - they can sometimes require permissions to restrict world read to private key data.

Would it make more sense for this to be `0640` instead? Also, should root have permission to *write* this key? Ideally this whole filesystem is read only (although that's a separate thing to the permissions we use)

Reply:
Why would we want to remove root the permission to write the key?

Reply:
I am not an expert on these file permissions, but [secrets-store-csi-driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver) uses [0644](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/23e09d50f85e45440d105711f994b5fc7350cbf2/pkg/secrets-store/nodeserver.go#L55).